src/vos/vos_policy.c

```diff
@@ -0,0 +1,61 @@
+/**
+ * (C) Copyright 2016-2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+/**
+ * Object placement policy
+ *  Policy definitions
+ *
+ */
+
+#include <daos_srv/policy.h>
+#include "vos_policy.h"
+
+/* policy functions definitions */
+
+/* policy based on io size
+ *
+ * Storage tier is determined using size of the IO operation and user specified
+ * per pool thresholds.
+ *
+ * This policy uses configurable parameters as size thresholds.
+ * If parameters are 0, default thresholds are used.
+ */
+static enum daos_media_type_t
+policy_io_size(struct vos_pool *pool, daos_iod_type_t type, daos_size_t size)
+{
+	uint32_t		scm_threshold;
+
+	if (pool->vp_vea_info == NULL)
+		return DAOS_MEDIA_SCM;
+
+	scm_threshold = pool->vp_policy_desc.params[0] > 0 ?
+		pool->vp_policy_desc.params[0] : VOS_POLICY_SCM_THRESHOLD;
```

Comment:
I didn't see any sanity check code on the threshold value. We need to reject invalid parameters and return error somewhere (at least in io engine, we'd ensure that no insane parameter value is used).

Reply:
I added a basic check in parse_param_val(), it assumed value should be >=0 and <= MAX32_UINT